### PR TITLE
API: Show activity notifications

### PR DESCRIPTION
**Pull request by annando**
The new API does support notificarions for reshares and likes - which we hadn't supported yet.

There is now a new table that will eventually replace the old notification system - but not in the current RC phase.

Now the new table is just used in parallel to the old system.

### CODE REVIEW

File: src/Factory/Api/Mastodon/Notification.php [hunk 1]
```diff
@@ -24,23 +24,18 @@
...
 use Friendica\DI;
-use Friendica\Model\Contact;
-use Friendica\Model\Notification as ModelNotification;
+use Friendica\Model\Post;
+use Friendica\Model\Verb;
+use Friendica\Protocol\Activity;
 
 class Notification extends BaseFactory
 {
 	public function createFromNotifyId(int $id)
```
Comment (MrPetovan) [inline src/Factory/Api/Mastodon/Notification.php]: Please rename to match the new table's name.
```suggestion
	public function createFromNotificationId(int $id)
```

File: src/Model/Post/UserNotification.php [hunk 2]
```diff
@@ -236,6 +266,32 @@ private static function setNotificationForUser(array $item, int $uid)
...
+			'created' => DateTimeFormat::utcNow(),
+		];
+
+		if ($item['gravity'] == GRAVITY_ACTIVITY) {
+			$fields['target-uri-id'] = $item['thr-parent-id'];
+		} else {
+			$fields['target-uri-id'] = $item['uri-id'];
+		}
+
+		dba::insert('notification', $fields);
```
Comment (MrPetovan) [inline src/Model/Post/UserNotification.php]: Please use the correct capitalization.
```suggestion
		DBA::insert('notification', $fields);
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Model/Post/UserNotification.php (hunk 2 above)
Comment (annando) [reply src/Model/Post/UserNotification.php]: Done

File: src/Factory/Api/Mastodon/Notification.php (hunk 1 above)
Comment (annando) [reply src/Factory/Api/Mastodon/Notification.php]: Done

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**